In [35]:
import pandas as pd
import sqlite3

In [36]:
# Open up a connection
conn = sqlite3.connect("zippedData/im.db")
# Initialize a cursor
cur = conn.cursor()

In [37]:
cur.execute("""SELECT name FROM sqlite_master WHERE type = 'table';""")
# Fetch the result and store it in table_names
table_names = cur.fetchall()
table_names

[('movie_basics',),
 ('directors',),
 ('known_for',),
 ('movie_akas',),
 ('movie_ratings',),
 ('persons',),
 ('principals',),
 ('writers',)]

In [38]:
movie_basics = pd.read_sql("""
SELECT * FROM movie_basics
""", conn)

movie_basics

movie_id                                primary_title  \
0       tt0063540                                    Sunghursh   
1       tt0066787              One Day Before the Rainy Season   
2       tt0069049                   The Other Side of the Wind   
3       tt0069204                              Sabse Bada Sukh   
4       tt0100275                     The Wandering Soap Opera   
...           ...                                          ...   
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
0                                         Sunghursh        2013   
1                                   Ashad Ka Ek Din        2019   
2                        The Other Side of the Wind        2018   
3                                   Sabse Bada Sukh        2018   
4                             La Telenovela Errante        2017   
...                                             ...         ...   
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes                genres  
0                 175.0    Action,Crime,Drama  
1                 114.0       Biography,Drama  
2                 122.0                 Drama  
3                   NaN          Comedy,Drama  
4                  80.0  Comedy,Drama,Fantasy  
...                 ...                   ...  
146139            123.0                 Drama  
146140              NaN           Documentary  
146141              NaN                Comedy  
146142            116.0                  None  
146143              NaN           Documentary  

[146144 rows x 6 columns]

In [39]:
directors = pd.read_sql("""
SELECT * FROM directors
""", conn)

directors

movie_id   person_id
0       tt0285252   nm0899854
1       tt0462036   nm1940585
2       tt0835418   nm0151540
3       tt0835418   nm0151540
4       tt0878654   nm0089502
...           ...         ...
291169  tt8999974  nm10122357
291170  tt9001390   nm6711477
291171  tt9001494  nm10123242
291172  tt9001494  nm10123248
291173  tt9004986   nm4993825

[291174 rows x 2 columns]

In [40]:
known_for = pd.read_sql("""
SELECT * FROM known_for
""", conn)

known_for

person_id   movie_id
0        nm0061671  tt0837562
1        nm0061671  tt2398241
2        nm0061671  tt0844471
3        nm0061671  tt0118553
4        nm0061865  tt0896534
...            ...        ...
1638255  nm9990690  tt9090932
1638256  nm9990690  tt8737130
1638257  nm9991320  tt8734436
1638258  nm9991320  tt9615610
1638259  nm9993380  tt8743182

[1638260 rows x 2 columns]

In [41]:
movie_akas = pd.read_sql("""
SELECT * FROM movie_akas
""", conn)

movie_akas

movie_id  ordering                                    title region  \
0       tt0369610        10                            Джурасик свят     BG   
1       tt0369610        11                        Jurashikku warudo     JP   
2       tt0369610        12  Jurassic World: O Mundo dos Dinossauros     BR   
3       tt0369610        13                  O Mundo dos Dinossauros     BR   
4       tt0369610        14                           Jurassic World     FR   
...           ...       ...                                      ...    ...   
331698  tt9827784         2                       Sayonara kuchibiru   None   
331699  tt9827784         3                            Farewell Song    XWW   
331700  tt9880178         1                              La atención   None   
331701  tt9880178         2                              La atención     ES   
331702  tt9880178         3                            The Attention    XWW   

       language        types   attributes  is_original_title  
0            bg         None         None                0.0  
1          None  imdbDisplay         None                0.0  
2          None  imdbDisplay         None                0.0  
3          None         None  short title                0.0  
4          None  imdbDisplay         None                0.0  
...         ...          ...          ...                ...  
331698     None     original         None                1.0  
331699       en  imdbDisplay         None                0.0  
331700     None     original         None                1.0  
331701     None         None         None                0.0  
331702       en  imdbDisplay         None                0.0  

[331703 rows x 8 columns]

In [42]:
movie_ratings = pd.read_sql("""
SELECT * FROM movie_ratings
""", conn)

movie_ratings

movie_id  averagerating  numvotes
0      tt10356526            8.3        31
1      tt10384606            8.9       559
2       tt1042974            6.4        20
3       tt1043726            4.2     50352
4       tt1060240            6.5        21
...           ...            ...       ...
73851   tt9805820            8.1        25
73852   tt9844256            7.5        24
73853   tt9851050            4.7        14
73854   tt9886934            7.0         5
73855   tt9894098            6.3       128

[73856 rows x 3 columns]

In [43]:
persons = pd.read_sql("""
SELECT * FROM persons
""", conn)

persons

person_id         primary_name  birth_year  death_year  \
0       nm0061671    Mary Ellen Bauder         NaN         NaN   
1       nm0061865         Joseph Bauer         NaN         NaN   
2       nm0062070           Bruce Baum         NaN         NaN   
3       nm0062195         Axel Baumann         NaN         NaN   
4       nm0062798          Pete Baxter         NaN         NaN   
...           ...                  ...         ...         ...   
606643  nm9990381         Susan Grobes         NaN         NaN   
606644  nm9990690          Joo Yeon So         NaN         NaN   
606645  nm9991320       Madeline Smith         NaN         NaN   
606646  nm9991786  Michelle Modigliani         NaN         NaN   
606647  nm9993380       Pegasus Envoyé         NaN         NaN   

                                      primary_profession  
0              miscellaneous,production_manager,producer  
1             composer,music_department,sound_department  
2                             miscellaneous,actor,writer  
3       camera_department,cinematographer,art_department  
4       production_designer,art_department,set_decorator  
...                                                  ...  
606643                                           actress  
606644                                           actress  
606645                                           actress  
606646                                          producer  
606647                             director,actor,writer  

[606648 rows x 5 columns]

In [44]:
principals = pd.read_sql("""
SELECT * FROM principals
""", conn)

principals

movie_id  ordering   person_id  category       job  \
0        tt0111414         1   nm0246005     actor      None   
1        tt0111414         2   nm0398271  director      None   
2        tt0111414         3   nm3739909  producer  producer   
3        tt0323808        10   nm0059247    editor      None   
4        tt0323808         1   nm3579312   actress      None   
...            ...       ...         ...       ...       ...   
1028181  tt9692684         1   nm0186469     actor      None   
1028182  tt9692684         2   nm4929530      self      None   
1028183  tt9692684         3  nm10441594  director      None   
1028184  tt9692684         4   nm6009913    writer    writer   
1028185  tt9692684         5  nm10441595  producer  producer   

                   characters  
0                 ["The Man"]  
1                        None  
2                        None  
3                        None  
4            ["Beth Boothby"]  
...                       ...  
1028181  ["Ebenezer Scrooge"]  
1028182   ["Herself","Regan"]  
1028183                  None  
1028184                  None  
1028185                  None  

[1028186 rows x 6 columns]

In [45]:
writers = pd.read_sql("""
SELECT * FROM writers
""", conn)

writers

movie_id   person_id
0       tt0285252   nm0899854
1       tt0438973   nm0175726
2       tt0438973   nm1802864
3       tt0462036   nm1940585
4       tt0835418   nm0310087
...           ...         ...
255868  tt8999892  nm10122246
255869  tt8999974  nm10122357
255870  tt9001390   nm6711477
255871  tt9004986   nm4993825
255872  tt9010172   nm8352242

[255873 rows x 2 columns]

In [46]:
# Read all csv/tsv files and create dataframes
movie_gross = pd.read_csv("zippedData/bom.movie_gross.csv.gz")
movie_info = pd.read_csv("zippedData/rt.movie_info.tsv.gz", sep='\t')
reviews = pd.read_csv("zippedData/rt.reviews.tsv.gz", sep='\t', encoding='latin1')
tmbd_movies = pd.read_csv("zippedData/tmdb.movies.csv.gz")
movie_budgets = pd.read_csv("zippedData/tn.movie_budgets.csv.gz")


In this section, we are reading several CSV and TSV files related to movies and their financial information, reviews, budgets, and more. Let's break down the process and explain what each dataset represents and what the output is expected to be.

1. Movie Gross (movie_gross)

We begin by reading the bom.movie_gross.csv.gz file, which contains information about the box office gross earnings of movies. This dataset will provide insights into how well movies performed at the box office, which is crucial for analyzing movie success.

In [47]:
movie_gross

title      studio  domestic_gross  \
0                                     Toy Story 3          BV     415000000.0   
1                      Alice in Wonderland (2010)          BV     334200000.0   
2     Harry Potter and the Deathly Hallows Part 1          WB     296000000.0   
3                                       Inception          WB     292600000.0   
4                             Shrek Forever After        P/DW     238700000.0   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.          6200.0   
3383                  Edward II (2018 re-release)          FM          4800.0   
3384                                     El Pacto        Sony          2500.0   
3385                                     The Swan  Synergetic          2400.0   
3386                            An Actor Prepares       Grav.          1700.0   

     foreign_gross  year  
0        652000000  2010  
1        691300000  2010  
2        664300000  2010  
3        535700000  2010  
4        513900000  2010  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[3387 rows x 5 columns]

We use pandas.read_csv() to read the file.

The file is compressed in the .gz format, and pandas handles it seamlessly.

This dataset includes columns such as:

 - movie: Name or title of the movie.

 - domestic_gross: Total earnings in the domestic market (usually the U.S.).

 - international_gross: Total earnings in international markets.

 - worldwide_gross: Total worldwide earnings (sum of domestic and international).

When we display the contents of movie_gross using movie_gross, it will show the first few rows of the dataset, allowing us to examine the gross earnings for each movie.



## Movie Info
The rt.reviews.tsv.gz file contains movie reviews, including the text of the review itself and associated metadata such as the rating or sentiment.This is a TSV (Tab-Separated Values) file, so we specify sep='\t' to let pandas know the delimiter.

The columns typically include:

id: A unique identifier for each movie.

title: Movie title.

year: Release year.

genres: Movie genres.

rating: Movie rating (such as from Rotten Tomatoes or IMDb).

We can use this dataset to gain insights into the movies' genres, release years, and ratings.

In [48]:
movie_info

id                                           synopsis rating  \
0        1  This gritty, fast-paced, and innovative police...      R   
1        3  New York City, not-too-distant-future: Eric Pa...      R   
2        5  Illeana Douglas delivers a superb performance ...      R   
3        6  Michael Douglas runs afoul of a treacherous su...      R   
4        7                                                NaN     NR   
...    ...                                                ...    ...   
1555  1996  Forget terrorists or hijackers -- there's a ha...      R   
1556  1997  The popular Saturday Night Live sketch was exp...     PG   
1557  1998  Based on a novel by Richard Powell, when the l...      G   
1558  1999  The Sandlot is a coming-of-age story about a g...     PG   
1559  2000  Suspended from the force, Paris cop Hubert is ...      R   

                                                  genre            director  \
0                   Action and Adventure|Classics|Drama    William Friedkin   
1                     Drama|Science Fiction and Fantasy    David Cronenberg   
2                     Drama|Musical and Performing Arts      Allison Anders   
3                            Drama|Mystery and Suspense      Barry Levinson   
4                                         Drama|Romance      Rodney Bennett   
...                                                 ...                 ...   
1555   Action and Adventure|Horror|Mystery and Suspense                 NaN   
1556                 Comedy|Science Fiction and Fantasy        Steve Barron   
1557  Classics|Comedy|Drama|Musical and Performing Arts      Gordon Douglas   
1558    Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   
1559  Action and Adventure|Art House and Internation...                 NaN   

                                                writer  theater_date  \
0                                       Ernest Tidyman   Oct 9, 1971   
1                         David Cronenberg|Don DeLillo  Aug 17, 2012   
2                                       Allison Anders  Sep 13, 1996   
3                      Paul Attanasio|Michael Crichton   Dec 9, 1994   
4                                         Giles Cooper           NaN   
...                                                ...           ...   
1555                                               NaN  Aug 18, 2006   
1556  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1557                                               NaN   Jan 1, 1962   
1558                  David Mickey Evans|Robert Gunter   Apr 1, 1993   
1559                                        Luc Besson  Sep 27, 2001   

          dvd_date currency  box_office      runtime             studio  
0     Sep 25, 2001      NaN         NaN  104 minutes                NaN  
1      Jan 1, 2013        $     600,000  108 minutes  Entertainment One  
2     Apr 18, 2000      NaN         NaN  116 minutes                NaN  
3     Aug 27, 1997      NaN         NaN  128 minutes                NaN  
4              NaN      NaN         NaN  200 minutes                NaN  
...            ...      ...         ...          ...                ...  
1555   Jan 2, 2007        $  33,886,034  106 minutes    New Line Cinema  
1556  Apr 17, 2001      NaN         NaN   88 minutes  Paramount Vantage  
1557  May 11, 2004      NaN         NaN  111 minutes                NaN  
1558  Jan 29, 2002      NaN         NaN  101 minutes                NaN  
1559  Feb 11, 2003      NaN         NaN   94 minutes  Columbia Pictures  

[1560 rows x 12 columns]

This is a TSV (Tab-Separated Values) file, so we specify sep='\t' to let pandas know the delimiter.

The columns typically include:

- id: Unique movie identifier (used for merging with other datasets).

- synopsis: Summary of the plot.

- rating: MPAA rating (e.g., PG-13, R).

- genre: Genre(s) of the movie.

- director: Director(s) of the movie.

- writer: Writer(s) of the movie.

- theater_date: Theater release date.

- dvd_date: DVD release date.

- currency: Currency used in box office numbers.

- box_office: Box office earnings.

- runtime: Movie length (in minutes).

- studio: Producing studio.

  
We can use this dataset to gain insights into the movies' genres, release years, and ratings.

## Reviews

The rt.reviews.tsv.gz file contains movie reviews, including the text of the review itself and associated metadata such as the rating or sentiment.

In [49]:
reviews

id                                             review rating   fresh  \
0         3  A distinctly gallows take on contemporary fina...    3/5   fresh   
1         3  It's an allegory in search of a meaning that n...    NaN  rotten   
2         3  ... life lived in a bubble in financial dealin...    NaN   fresh   
3         3  Continuing along a line introduced in last yea...    NaN   fresh   
4         3             ... a perverse twist on neorealism...     NaN   fresh   
...     ...                                                ...    ...     ...   
54427  2000  The real charm of this trifle is the deadpan c...    NaN   fresh   
54428  2000                                                NaN    1/5  rotten   
54429  2000                                                NaN    2/5  rotten   
54430  2000                                                NaN  2.5/5  rotten   
54431  2000                                                NaN    3/5   fresh   

                  critic  top_critic         publisher                date  
0             PJ Nabarro           0   Patrick Nabarro   November 10, 2018  
1         Annalee Newitz           0           io9.com        May 23, 2018  
2           Sean Axmaker           0  Stream on Demand     January 4, 2018  
3          Daniel Kasman           0              MUBI   November 16, 2017  
4                    NaN           0      Cinema Scope    October 12, 2017  
...                  ...         ...               ...                 ...  
54427      Laura Sinagra           1     Village Voice  September 24, 2002  
54428  Michael Szymanski           0        Zap2it.com  September 21, 2005  
54429       Emanuel Levy           0   EmanuelLevy.Com       July 17, 2005  
54430   Christopher Null           0    Filmcritic.com   September 7, 2003  
54431    Nicolas Lacroix           0      Showbizz.net   November 12, 2002  

[54432 rows x 8 columns]

Again, this is a TSV file, so we use sep='\t' as the delimiter.

The encoding is set to 'latin1' because the file may contain special characters not supported by the default UTF-8 encoding.

Columns might include:

- id: Movie ID (to link with movie_info).

- review: Full text of the review.

- rating: Critic rating (if available).

- fresh: Whether the review is "Fresh" or "Rotten".

- critic: Name of the critic.

- top_critic: Boolean flag for top-tier critics.

- publisher: Publication the review came from.

- date: Date the review was published.

This dataset is ideal for sentiment analysis and understanding how critics responded to specific films.

## TMBD Movies

Next, we load the tmdb.movies.csv.gz file, which contains detailed movie data from The Movie Database (TMDb), including metadata such as runtime, budget, revenue, and more.

In [50]:
tmbd_movies

Unnamed: 0            genre_ids      id original_language  \
0               0      [12, 14, 10751]   12444                en   
1               1  [14, 12, 16, 10751]   10191                en   
2               2        [12, 28, 878]   10138                en   
3               3      [16, 35, 10751]     862                en   
4               4        [28, 878, 12]   27205                en   
...           ...                  ...     ...               ...   
26512       26512             [27, 18]  488143                en   
26513       26513             [18, 53]  485975                en   
26514       26514         [14, 28, 12]  381231                en   
26515       26515      [10751, 12, 28]  366854                en   
26516       26516             [53, 27]  309885                en   

                                     original_title  popularity release_date  \
0      Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                          How to Train Your Dragon      28.734   2010-03-26   
2                                        Iron Man 2      28.515   2010-05-07   
3                                         Toy Story      28.005   1995-11-22   
4                                         Inception      27.920   2010-07-16   
...                                             ...         ...          ...   
26512                         Laboratory Conditions       0.600   2018-10-13   
26513                               _EXHIBIT_84xxx_       0.600   2018-05-01   
26514                                  The Last One       0.600   2018-10-01   
26515                                  Trailer Made       0.600   2018-06-22   
26516                                    The Church       0.600   2018-10-05   

                                              title  vote_average  vote_count  
0      Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                          How to Train Your Dragon           7.7        7610  
2                                        Iron Man 2           6.8       12368  
3                                         Toy Story           7.9       10174  
4                                         Inception           8.3       22186  
...                                             ...           ...         ...  
26512                         Laboratory Conditions           0.0           1  
26513                               _EXHIBIT_84xxx_           0.0           1  
26514                                  The Last One           0.0           1  
26515                                  Trailer Made           0.0           1  
26516                                    The Church           0.0           1  

[26517 rows x 10 columns]

This file contains CSV data, and pandas will read it directly.

The columns in this dataset might include:

- Unnamed: 0: Index column (can be dropped).

- genre_ids: Genre codes (numeric).

- id: Movie ID.

- original_language: Language of the movie.

- original_title: Original title (before translation).

- popularity: Popularity score.

- release_date: Date of release.

- title: Movie title.

- vote_average: Average viewer rating.

- vote_count: Number of votes.

## Movie Budgets

Finally, we load the tn.movie_budgets.csv.gz file, which contains the movie production budgets.

In [51]:
movie_budgets

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

     production_budget domestic_gross worldwide_gross  
0         $425,000,000   $760,507,625  $2,776,345,279  
1         $410,600,000   $241,063,875  $1,045,663,875  
2         $350,000,000    $42,762,350    $149,762,350  
3         $330,600,000   $459,005,868  $1,403,013,963  
4         $317,000,000   $620,181,382  $1,316,721,747  
...                ...            ...             ...  
5777            $7,000             $0              $0  
5778            $6,000        $48,482        $240,495  
5779            $5,000         $1,338          $1,338  
5780            $1,400             $0              $0  
5781            $1,100       $181,041        $181,041  

[5782 rows x 6 columns]

This file is a CSV containing information about movie production budgets.

Columns typically include:

- id: Unique ID (can be created or cleaned from other columns if needed).

- release_date: The release date of the movie.

- movie: Title of the movie.

- production_budget: Cost of producing the film.

- domestic_gross: U.S. earnings.

- worldwide_gross: Global earnings.

This dataset enables financial analysis like ROI (Return on Investment), budgeting trends, and profit comparison across movies.


## Main Data Preview.
We perform an initial data preview and summary of the movie_basics DataFrame to understand its structure and content.

In [52]:
movie_basics.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   movie_id         146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


Number of entries (146,144 movies).

Column names and data types (object, int64, float64).

Count of non-null values (useful for spotting missing data).



Why it's useful:

It helps quickly identify:

- Which columns may need cleaning (e.g., runtime_minutes and genres have missing values).

- The data types, so you know which columns can be used for calculations, grouping, or plotting.

In [53]:
movie_basics.describe()

start_year  runtime_minutes
count  146144.000000    114405.000000
mean     2014.621798        86.187247
std         2.733583       166.360590
min      2010.000000         1.000000
25%      2012.000000        70.000000
50%      2015.000000        87.000000
75%      2017.000000        99.000000
max      2115.000000     51420.000000

This gives statistical summaries for numeric columns only (start_year, runtime_minutes).

What it shows:

- count: Number of non-null values.

- mean: Average year and runtime.

- std: Standard deviation (how spread out the values are).

- min/max: Earliest/latest year and shortest/longest runtime.

- 25%/50%/75%: Percentiles for distribution insights.

Why it's useful:

- Identifies potential outliers (e.g., max runtime of 51420 minutes is suspicious).

- Gives a snapshot of how most movies are distributed in terms of release year and length.

- Helps with decisions like filtering extreme values, normalizing data, or plotting histograms.



For null original titles we will give it the primary title

## We will work on "movie_gross" as our base dataset**

In [54]:
movie_gross.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


This command gives a summary of the data structure:

- Rows: 3,387 movies.

- Columns: 5 (title, studio, domestic_gross, foreign_gross, year).

Non-null values:

Most columns are nearly complete, except:

- foreign_gross: only ~60% of entries are present (2,037 non-null).

- studio: 5 missing values.

- domestic_gross: 28 missing values.

Data types:

- domestic_gross is float64 (numeric).

- foreign_gross is still an object, which may contain strings or improperly formatted numbers and may require cleaning.

It highlights columns needing cleaning or conversion (e.g., convert foreign_gross to numeric), and shows overall dataset completeness.



In [55]:
movie_gross.describe()

domestic_gross         year
count    3.359000e+03  3387.000000
mean     2.874585e+07  2013.958075
std      6.698250e+07     2.478141
min      1.000000e+02  2010.000000
25%      1.200000e+05  2012.000000
50%      1.400000e+06  2014.000000
75%      2.790000e+07  2016.000000
max      9.367000e+08  2018.000000

This provides summary statistics for numeric columns (domestic_gross and year):

- domestic_gross:

- Count: 3,359 values (some missing).

- Mean: ~$28.7 million.

- Min: $100 (likely a data error or limited release).

- Max: ~$936.7 million (blockbusters).

Large standard deviation shows high variance in box office performance.

- year:

Range: 2010 to 2018.

Even distribution across the time period (median year is 2014).


It helps us understand value distributions, detect outliers (like very low or high gross), and plan how to handle those in analysis.



- This line below converts all movie titles to lowercase in the title column of the movie_gross DataFrame.

The method .str.lower() is applied to each string in the title column.

It standardizes the text format, which is useful for:

Merging datasets (e.g., avoiding mismatches due to capitalization).

Searching/filtering titles consistently.

Avoiding duplicates caused by case differences.

In [56]:
movie_gross['title'] = movie_gross['title'].str.lower()

In [57]:
movie_gross

title      studio  domestic_gross  \
0                                     toy story 3          BV     415000000.0   
1                      alice in wonderland (2010)          BV     334200000.0   
2     harry potter and the deathly hallows part 1          WB     296000000.0   
3                                       inception          WB     292600000.0   
4                             shrek forever after        P/DW     238700000.0   
...                                           ...         ...             ...   
3382                                    the quake       Magn.          6200.0   
3383                  edward ii (2018 re-release)          FM          4800.0   
3384                                     el pacto        Sony          2500.0   
3385                                     the swan  Synergetic          2400.0   
3386                            an actor prepares       Grav.          1700.0   

     foreign_gross  year  
0        652000000  2010  
1        691300000  2010  
2        664300000  2010  
3        535700000  2010  
4        513900000  2010  
...            ...   ...  
3382           NaN  2018  
3383           NaN  2018  
3384           NaN  2018  
3385           NaN  2018  
3386           NaN  2018  

[3387 rows x 5 columns]

You're showing a preview of the movie_gross DataFrame after this transformation. Here's what we can observe:

Columns:

- title – Now all lowercase (e.g., "Toy Story 3" → "toy story 3").

- studio – Abbreviations of production companies (e.g., WB for Warner Bros).

- domestic_gross – Revenue earned in the U.S. (numeric).

- foreign_gross – Revenue from outside the U.S. (some still NaN or strings – may need cleaning).

- year – Year of release.

Rows:

- Each row represents a movie and its financial performance.

- Later rows show smaller box office numbers and missing foreign_gross values.



In [62]:
movie_info.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1560 entries, 0 to 1559
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   id            1560 non-null   int64 
 1   synopsis      1498 non-null   object
 2   rating        1557 non-null   object
 3   genre         1552 non-null   object
 4   director      1361 non-null   object
 5   writer        1111 non-null   object
 6   theater_date  1201 non-null   object
 7   dvd_date      1201 non-null   object
 8   currency      340 non-null    object
 9   box_office    340 non-null    object
 10  runtime       1530 non-null   object
 11  studio        494 non-null    object
dtypes: int64(1), object(11)
memory usage: 146.4+ KB


This command provides a concise summary of the structure and completeness of the movie_info DataFrame. It's a key part of Exploratory Data Analysis (EDA) to assess the dataset before doing any transformations or merging.

- This dataset is rich in movie metadata, but has many missing values in critical fields like box_office, writer, studio, and release dates.

- Some numeric-looking columns (runtime, box_office) are typed as object and will likely need conversion to numeric types.

- The id column is complete and likely used to join with other Rotten Tomatoes datasets (like reviews).

Why does this matter?

- Helps identify which columns need cleaning, type conversion, or imputation.

- Lets you decide which columns are reliable for analysis or merging with other datasets.

- Guides your data cleaning workflow (e.g., converting box_office to float, parsing dates, filling missing genres or runtimes).

### We then rename the rating column in the movie_info DataFrame to age_rating:

- Makes the column's meaning explicit (MPAA-style age classification).

- Prevents column name collisions during merges.

- Improves code readability and reduces bugs downstream.

In [63]:
# Rename rating column for clarity when joining
movie_info.rename(columns= {'rating':'age_rating'}, inplace=True)
movie_info

id                                           synopsis age_rating  \
0        1  This gritty, fast-paced, and innovative police...          R   
1        3  New York City, not-too-distant-future: Eric Pa...          R   
2        5  Illeana Douglas delivers a superb performance ...          R   
3        6  Michael Douglas runs afoul of a treacherous su...          R   
4        7                                                NaN         NR   
...    ...                                                ...        ...   
1555  1996  Forget terrorists or hijackers -- there's a ha...          R   
1556  1997  The popular Saturday Night Live sketch was exp...         PG   
1557  1998  Based on a novel by Richard Powell, when the l...          G   
1558  1999  The Sandlot is a coming-of-age story about a g...         PG   
1559  2000  Suspended from the force, Paris cop Hubert is ...          R   

                                                  genre            director  \
0                   Action and Adventure|Classics|Drama    William Friedkin   
1                     Drama|Science Fiction and Fantasy    David Cronenberg   
2                     Drama|Musical and Performing Arts      Allison Anders   
3                            Drama|Mystery and Suspense      Barry Levinson   
4                                         Drama|Romance      Rodney Bennett   
...                                                 ...                 ...   
1555   Action and Adventure|Horror|Mystery and Suspense                 NaN   
1556                 Comedy|Science Fiction and Fantasy        Steve Barron   
1557  Classics|Comedy|Drama|Musical and Performing Arts      Gordon Douglas   
1558    Comedy|Drama|Kids and Family|Sports and Fitness  David Mickey Evans   
1559  Action and Adventure|Art House and Internation...                 NaN   

                                                writer  theater_date  \
0                                       Ernest Tidyman   Oct 9, 1971   
1                         David Cronenberg|Don DeLillo  Aug 17, 2012   
2                                       Allison Anders  Sep 13, 1996   
3                      Paul Attanasio|Michael Crichton   Dec 9, 1994   
4                                         Giles Cooper           NaN   
...                                                ...           ...   
1555                                               NaN  Aug 18, 2006   
1556  Terry Turner|Tom Davis|Dan Aykroyd|Bonnie Turner  Jul 23, 1993   
1557                                               NaN   Jan 1, 1962   
1558                  David Mickey Evans|Robert Gunter   Apr 1, 1993   
1559                                        Luc Besson  Sep 27, 2001   

          dvd_date currency  box_office      runtime             studio  
0     Sep 25, 2001      NaN         NaN  104 minutes                NaN  
1      Jan 1, 2013        $     600,000  108 minutes  Entertainment One  
2     Apr 18, 2000      NaN         NaN  116 minutes                NaN  
3     Aug 27, 1997      NaN         NaN  128 minutes                NaN  
4              NaN      NaN         NaN  200 minutes                NaN  
...            ...      ...         ...          ...                ...  
1555   Jan 2, 2007        $  33,886,034  106 minutes    New Line Cinema  
1556  Apr 17, 2001      NaN         NaN   88 minutes  Paramount Vantage  
1557  May 11, 2004      NaN         NaN  111 minutes                NaN  
1558  Jan 29, 2002      NaN         NaN  101 minutes                NaN  
1559  Feb 11, 2003      NaN         NaN   94 minutes  Columbia Pictures  

[1560 rows x 12 columns]

## We then work on the movie_budgets table to extract the production_budget

In [64]:

movie_budgets.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5782 entries, 0 to 5781
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   id                 5782 non-null   int64 
 1   release_date       5782 non-null   object
 2   movie              5782 non-null   object
 3   production_budget  5782 non-null   object
 4   domestic_gross     5782 non-null   object
 5   worldwide_gross    5782 non-null   object
dtypes: int64(1), object(5)
memory usage: 271.2+ KB


This command gives a summary of the structure of the movie_budgets DataFrame.

Output Explanation:

- Rows: 5,782 entries — each row represents a single movie.

- Columns: 6 columns with various movie budget and revenue data.



In [67]:
movie_budgets

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       Avatar   
1      2  May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
2      3   Jun 7, 2019                                 Dark Phoenix   
3      4   May 1, 2015                      Avengers: Age of Ultron   
4      5  Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       Red 11   
5778  79   Apr 2, 1999                                    Following   
5779  80  Jul 13, 2005                Return to the Land of Wonders   
5780  81  Sep 29, 2015                         A Plague So Pleasant   
5781  82   Aug 5, 2005                            My Date With Drew   

     production_budget domestic_gross worldwide_gross  
0         $425,000,000   $760,507,625  $2,776,345,279  
1         $410,600,000   $241,063,875  $1,045,663,875  
2         $350,000,000    $42,762,350    $149,762,350  
3         $330,600,000   $459,005,868  $1,403,013,963  
4         $317,000,000   $620,181,382  $1,316,721,747  
...                ...            ...             ...  
5777            $7,000             $0              $0  
5778            $6,000        $48,482        $240,495  
5779            $5,000         $1,338          $1,338  
5780            $1,400             $0              $0  
5781            $1,100       $181,041        $181,041  

[5782 rows x 6 columns]

These values are currently strings, so you’ll need to:

Remove $ and ,

Convert them to integers or floats for analysis or plotting

## Cleaning production_budget column:


The lines below remove unwanted characters (commas and dollar signs) from the production_budget column.

The original values were like "$425,000,000", which are strings.

You cannot do numerical operations (e.g., sum, average) on those until they’re cleaned.





This line provides summary statistics for all numeric columns, especially production_budget (since it was just converted).

In [68]:
movie_budgets['production_budget'] = movie_budgets['production_budget'].str.replace(",","")
movie_budgets['production_budget'] = movie_budgets['production_budget'].str.replace("$","")
movie_budgets['production_budget'] = movie_budgets['production_budget'].astype(int)

In [69]:
movie_budgets['movie'] = movie_budgets['movie'].str.lower()

Converting to integer:

- Converts the cleaned strings to integers so we can perform numeric operations like .describe(), plotting, or math.

- Converts all movie titles to lowercase to ensure consistency when comparing or merging with other datasets.

- This avoids issues where "Avatar" and "avatar" would be treated as different.



In [70]:
movie_budgets.describe()

id  production_budget
count  5782.000000       5.782000e+03
mean     50.372363       3.158776e+07
std      28.821076       4.181208e+07
min       1.000000       1.100000e+03
25%      25.000000       5.000000e+06
50%      50.000000       1.700000e+07
75%      75.000000       4.000000e+07
max     100.000000       4.250000e+08

This line provides summary statistics for all numeric columns, especially production_budget (since it was just converted).





- The average production budget is around $31M.

- The minimum budget is just over $1,000 (very small indie films).

- The maximum is $425M (big-budget blockbusters like Avatar).

- The spread (std) shows a large variation in movie budgets.

In [71]:
movie_budgets

id  release_date                                        movie  \
0      1  Dec 18, 2009                                       avatar   
1      2  May 20, 2011  pirates of the caribbean: on stranger tides   
2      3   Jun 7, 2019                                 dark phoenix   
3      4   May 1, 2015                      avengers: age of ultron   
4      5  Dec 15, 2017            star wars ep. viii: the last jedi   
...   ..           ...                                          ...   
5777  78  Dec 31, 2018                                       red 11   
5778  79   Apr 2, 1999                                    following   
5779  80  Jul 13, 2005                return to the land of wonders   
5780  81  Sep 29, 2015                         a plague so pleasant   
5781  82   Aug 5, 2005                            my date with drew   

      production_budget domestic_gross worldwide_gross  
0             425000000   $760,507,625  $2,776,345,279  
1             410600000   $241,063,875  $1,045,663,875  
2             350000000    $42,762,350    $149,762,350  
3             330600000   $459,005,868  $1,403,013,963  
4             317000000   $620,181,382  $1,316,721,747  
...                 ...            ...             ...  
5777               7000             $0              $0  
5778               6000        $48,482        $240,495  
5779               5000         $1,338          $1,338  
5780               1400             $0              $0  
5781               1100       $181,041        $181,041  

[5782 rows x 6 columns]

### We will then work on cleaning the movie_basics table. We will combine it with movie_ratingstable on sql to form a different dataframe.

Loading individual tables into pandas:

In [72]:
# Analyzing the tables and load data into a pandas DataFrame
query1 = pd.read_sql_query( "SELECT * FROM movie_ratings", conn)

query2 = pd.read_sql_query("SELECT * FROM movie_basics" , conn)
print("Movie Ratings table \n",query1.head()) 
print("Movie Basics table \n",query2.head())

Movie Ratings table 
      movie_id  averagerating  numvotes
0  tt10356526            8.3        31
1  tt10384606            8.9       559
2   tt1042974            6.4        20
3   tt1043726            4.2     50352
4   tt1060240            6.5        21
Movie Basics table 
     movie_id                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          

query1: Loads the movie_ratings table (contains IMDb-like rating data).

query2: Loads the movie_basics table (contains core movie details like title, genre, year).

### SQL JOIN to combine both tables:

In [73]:
sql_dataset = pd.read_sql_query(
    """
SELECT 
    mb.movie_id,
    mb.original_title,
    mb.primary_title,
    mb.genres,
    mb.start_year,
    mr.averagerating,
    mr.numvotes,
    mb.runtime_minutes
FROM 
    movie_basics AS mb
LEFT JOIN 
    movie_ratings AS mr
ON 
    mb.movie_id = mr.movie_id;
    """, conn)
print(sql_dataset.head())

    movie_id              original_title                    primary_title  \
0  tt0063540                   Sunghursh                        Sunghursh   
1  tt0066787             Ashad Ka Ek Din  One Day Before the Rainy Season   
2  tt0069049  The Other Side of the Wind       The Other Side of the Wind   
3  tt0069204             Sabse Bada Sukh                  Sabse Bada Sukh   
4  tt0100275       La Telenovela Errante         The Wandering Soap Opera   

                 genres  start_year  averagerating  numvotes  runtime_minutes  
0    Action,Crime,Drama        2013            7.0      77.0            175.0  
1       Biography,Drama        2019            7.2      43.0            114.0  
2                 Drama        2018            6.9    4517.0            122.0  
3          Comedy,Drama        2018            6.1      13.0              NaN  
4  Comedy,Drama,Fantasy        2017            6.5     119.0             80.0  


This SQL query joins the two tables on the movie_id.

A LEFT JOIN ensures that all records from movie_basics are kept, even if there's no matching rating in movie_ratings.

The result is stored in sql_dataset.





- We are combining the metadata (title, genre, runtime) from movie_basics with popularity data (ratings, votes) from movie_ratings.

- This merged dataset sql_dataset is much more useful for analysis, since it gives both descriptive and performance-related features per movie.

It can now be used for:

- Analyzing rating trends by genre or year

- Visualizing runtime vs rating

- Filtering movies with a minimum number of votes, etc.

### We check the current dataset info:

In [74]:
sql_dataset.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 8 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   movie_id         146144 non-null  object 
 1   original_title   146123 non-null  object 
 2   primary_title    146144 non-null  object 
 3   genres           140736 non-null  object 
 4   start_year       146144 non-null  int64  
 5   averagerating    73856 non-null   float64
 6   numvotes         73856 non-null   float64
 7   runtime_minutes  114405 non-null  float64
dtypes: float64(3), int64(1), object(4)
memory usage: 8.9+ MB


In [75]:
# Drop null values
sql_dataset.dropna(inplace = True)

Removes all rows that contain any NaN (null/missing) values.

inplace=True means the changes are applied directly to sql_dataset without needing to assign it to a new variable.

In [76]:
sql_dataset.info()

<class 'pandas.core.frame.DataFrame'>
Index: 65720 entries, 0 to 146134
Data columns (total 8 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   movie_id         65720 non-null  object 
 1   original_title   65720 non-null  object 
 2   primary_title    65720 non-null  object 
 3   genres           65720 non-null  object 
 4   start_year       65720 non-null  int64  
 5   averagerating    65720 non-null  float64
 6   numvotes         65720 non-null  float64
 7   runtime_minutes  65720 non-null  float64
dtypes: float64(3), int64(1), object(4)
memory usage: 4.5+ MB


- Before dropping nulls:

Total entries: 146,144

Some columns (like averagerating, runtime_minutes, etc.) had missing values.

- After dropping nulls:
              
Only 65,720 entries remain.

All columns now have complete data (no missing values).

This means 80,424 rows were removed.

In [77]:
sql_dataset

movie_id              original_title  \
0       tt0063540                   Sunghursh   
1       tt0066787             Ashad Ka Ek Din   
2       tt0069049  The Other Side of the Wind   
4       tt0100275       La Telenovela Errante   
7       tt0137204             Joe Finds Grace   
...           ...                         ...   
146113  tt9911774   Padmavyuhathile Abhimanyu   
146114  tt9913056                Swarm Season   
146115  tt9913084            Diabolik sono io   
146122  tt9914286           Sokagin Çocuklari   
146134  tt9916160                  Drømmeland   

                          primary_title                      genres  \
0                             Sunghursh          Action,Crime,Drama   
1       One Day Before the Rainy Season             Biography,Drama   
2            The Other Side of the Wind                       Drama   
4              The Wandering Soap Opera        Comedy,Drama,Fantasy   
7                       Joe Finds Grace  Adventure,Animation,Comedy   
...                                 ...                         ...   
146113        Padmavyuhathile Abhimanyu                       Drama   
146114                     Swarm Season                 Documentary   
146115                 Diabolik sono io                 Documentary   
146122                Sokagin Çocuklari                Drama,Family   
146134                       Drømmeland                 Documentary   

        start_year  averagerating  numvotes  runtime_minutes  
0             2013            7.0      77.0            175.0  
1             2019            7.2      43.0            114.0  
2             2018            6.9    4517.0            122.0  
4             2017            6.5     119.0             80.0  
7             2017            8.1     263.0             83.0  
...            ...            ...       ...              ...  
146113        2019            8.4     365.0            130.0  
146114        2019            6.2       5.0             86.0  
146115        2019            6.2       6.0             75.0  
146122        2019            8.7     136.0             98.0  
146134        2019            6.5      11.0             72.0  

[65720 rows x 8 columns]

### Next step is cleaning the tmdb dataset to extract the release_month, popularity

In [78]:
tmbd_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Unnamed: 0         26517 non-null  int64  
 1   genre_ids          26517 non-null  object 
 2   id                 26517 non-null  int64  
 3   original_language  26517 non-null  object 
 4   original_title     26517 non-null  object 
 5   popularity         26517 non-null  float64
 6   release_date       26517 non-null  object 
 7   title              26517 non-null  object 
 8   vote_average       26517 non-null  float64
 9   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(3), object(5)
memory usage: 2.0+ MB


Displays the structure of the tmbd_movies DataFrame: number of entries, columns, non-null counts, and data types.

In [79]:
# Convert release date, data type
tmbd_movies['release_date'] = pd.to_datetime(tmbd_movies['release_date'], errors='coerce')

Converts the release_date column (likely a string) to proper datetime format.

errors='coerce' means invalid or missing dates will be set as NaT (Not a Time) instead of causing an error.

In [80]:
tmbd_movies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 26517 entries, 0 to 26516
Data columns (total 10 columns):
 #   Column             Non-Null Count  Dtype         
---  ------             --------------  -----         
 0   Unnamed: 0         26517 non-null  int64         
 1   genre_ids          26517 non-null  object        
 2   id                 26517 non-null  int64         
 3   original_language  26517 non-null  object        
 4   original_title     26517 non-null  object        
 5   popularity         26517 non-null  float64       
 6   release_date       26517 non-null  datetime64[ns]
 7   title              26517 non-null  object        
 8   vote_average       26517 non-null  float64       
 9   vote_count         26517 non-null  int64         
dtypes: datetime64[ns](1), float64(2), int64(3), object(4)
memory usage: 2.0+ MB


Confirms that release_date is now in datetime64 format and shows any new missing values (NaT).

In [81]:
# Extract month from release date
tmbd_movies['release_month'] = tmbd_movies['release_date'].dt.month
tmbd_movies['release_month']

0        11
1         3
2         5
3        11
4         7
         ..
26512    10
26513     5
26514    10
26515     6
26516    10
Name: release_month, Length: 26517, dtype: int32

Creates a new column release_month that contains just the month (1–12) of the movie's release.



In [82]:
tmbd_movies

Unnamed: 0            genre_ids      id original_language  \
0               0      [12, 14, 10751]   12444                en   
1               1  [14, 12, 16, 10751]   10191                en   
2               2        [12, 28, 878]   10138                en   
3               3      [16, 35, 10751]     862                en   
4               4        [28, 878, 12]   27205                en   
...           ...                  ...     ...               ...   
26512       26512             [27, 18]  488143                en   
26513       26513             [18, 53]  485975                en   
26514       26514         [14, 28, 12]  381231                en   
26515       26515      [10751, 12, 28]  366854                en   
26516       26516             [53, 27]  309885                en   

                                     original_title  popularity release_date  \
0      Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                          How to Train Your Dragon      28.734   2010-03-26   
2                                        Iron Man 2      28.515   2010-05-07   
3                                         Toy Story      28.005   1995-11-22   
4                                         Inception      27.920   2010-07-16   
...                                             ...         ...          ...   
26512                         Laboratory Conditions       0.600   2018-10-13   
26513                               _EXHIBIT_84xxx_       0.600   2018-05-01   
26514                                  The Last One       0.600   2018-10-01   
26515                                  Trailer Made       0.600   2018-06-22   
26516                                    The Church       0.600   2018-10-05   

                                              title  vote_average  vote_count  \
0      Harry Potter and the Deathly Hallows: Part 1           7.7       10788   
1                          How to Train Your Dragon           7.7        7610   
2                                        Iron Man 2           6.8       12368   
3                                         Toy Story           7.9       10174   
4                                         Inception           8.3       22186   
...                                             ...           ...         ...   
26512                         Laboratory Conditions           0.0           1   
26513                               _EXHIBIT_84xxx_           0.0           1   
26514                                  The Last One           0.0           1   
26515                                  Trailer Made           0.0           1   
26516                                    The Church           0.0           1   

       release_month  
0                 11  
1                  3  
2                  5  
3                 11  
4                  7  
...              ...  
26512             10  
26513              5  
26514             10  
26515              6  
26516             10  

[26517 rows x 11 columns]

Shows the full DataFrame including the new release_month column.

Why This Is Done

- Converting dates to datetime makes them easier to work with for analysis (e.g., grouping by month/year).

- Extracting the month allows you to analyze trends like:

- When most movies are released

- Which months perform best in revenue or ratings